sys/kern/pool.c

```diff
@@ -64,6 +64,12 @@ typedef struct pool_item {
   unsigned long pi_data[0] __aligned(PI_ALIGNMENT);
 } pool_item_t;
 
+/* Pool of pool_t objects. */
+static pool_t P_POOL[1];
+
+#define POOL_BOOTPAGE_CNT 1
+static alignas(PAGESIZE) uint8_t P_POOL_BOOTPAGE[POOL_BOOTPAGE_CNT][PAGESIZE];
```

Comment:
I'd much prefer to have it defined as:
```c
static alignas(PAGESIZE) uint8_t P_POOL_BOOTPAGE[PAGESIZE*2];
```

Reply:
In quarantine PR, `POOL_BOOTPAGE_CNT` will have to be larger if KASAN is enabled (quarantine increases `sizeof(pool_t)` significantly), that's why I introduced `POOL_BOOTPAGE_CNT`.

So, would:
```
#define POOL_BOOTPAGE_CNT 1
static alignas(PAGESIZE) uint8_t P_POOL_BOOTPAGE[PAGESIZE * POOL_BOOTPAGE_CNT];
```
be OK?

And:
```
#ifdef KASAN
#define POOL_BOOTPAGE_CNT some-bigger-number
#else
#define POOL_BOOTPAGE_CNT 1
#endif /* !KASAN */
static alignas(PAGESIZE) uint8_t P_POOL_BOOTPAGE[PAGESIZE * POOL_BOOTPAGE_CNT];
```
in the future.

Reply:
It's no use of trying to give the structure to a workaround. I'll finally solve the problem with `M_NOGROW` at some point.

Reply:
So what do you expect me to do here?

PS I think that size of `P_POOL_BOOTPAGE` isn't related to `M_NOGROW`. During `pool_bootstrap` we are calling `INVOKE_CTORS(pool_ctor_table)` which allocates various memory pools out of "master pool" `P_POOL`. As this happens before `kmem_bootstrap`, "master pool" has to use static memory.

Reply:
Fixed in https://github.com/cahirwpz/mimiker/pull/693/commits/e4eca308c7799b49baf32adb53ddf638f6c5d127